semantics/src/traversal/functions.rs

```diff
@@ -106,14 +106,28 @@ fn func_stmt(
         fe::FuncStmt::While { .. } => unimplemented!(),
         fe::FuncStmt::If { .. } => unimplemented!(),
         fe::FuncStmt::Assert { .. } => unimplemented!(),
-        fe::FuncStmt::Expr { .. } => unimplemented!(),
+        fe::FuncStmt::Expr { .. } => expr(scope, context, stmt),
         fe::FuncStmt::Pass => unimplemented!(),
         fe::FuncStmt::Break => unimplemented!(),
         fe::FuncStmt::Continue => unimplemented!(),
         fe::FuncStmt::Revert => Ok(()),
     }
 }
 
+fn expr(
+    scope: Shared<FunctionScope>,
+    context: Shared<Context>,
+    stmt: &Spanned<fe::FuncStmt>,
+) -> Result<(), SemanticError> {
+    if let fe::FuncStmt::Expr { value } = &stmt.node {
+        let spanned = spanned_expression(&stmt.span, value);
+        // TODO: Whats the reason the expr isn't spanned?
+        let _attributes = expressions::expr(scope, context, &spanned)?;
```

Comment:
If we were to compile a function call statement that does not return a value, we would get a `NotAnExpression` error, since the `expr` method expects something that has some value.

I'm personally in favor of just treating call statements and call expressions as two separate node types and handling them in distinct branches of code, but arguments could be made in favor introducing a `void` type for situations like this.

Reply:
>If we were to compile a function call statement that does not return a value, we would get a NotAnExpression error,

Yeah, that makes a lot of sense. I didn't run into it because I accidentally gave `assign` a return value which the semantic pass did not complain about. I filed a separate issue for that https://github.com/ethereum/fe/issues/104

>I'm personally in favor of just treating call statements and call expressions as two separate node types and handling them in distinct branches of code

Yeah, I remember that but when I noticed that function call statements are hitting `fe::FuncStmt::Expr` I was scratching my head what else I might need to handle here. I guess I was expecting something like `fe::FuncStmt::Call` to exist.

>but arguments could be made in favor introducing a void type for situations like this

In general, I think that expression based programming is very powerful and I consider it one of the beauties of Rust to be so expression focused that pretty much everything (if/else, loops etc) is an expression. Expressions often tighten the rules and hence eliminate sources of bugs.

E.g. this simple if/else won't catch the typo in the assignment.
```
if something:
   some_var = "foo"
else:
   zome_var = "bar"
```

The same thing can not happen if a ternary expression was used.

```
some_var = "foo" if something else "bar"
```

What would be your main argument against introducing a `void` type?

Reply:
I started to address #104 in a separate PR https://github.com/ethereum/fe/pull/105/

Reply:
> What would be your main argument against introducing a void type?

I don't really have a strong argument against it and the more I think about it, the more I think we should add it. My initial hesitation just stemmed from adding another type variant, which we would then need to handle throughout the codebase. 

What do you think about using an empty tuple as our void/none type?

Reply:
>the more I think about it, the more I think we should add it

Cool :+1: 

>What do you think about using an empty tuple as our void/none type?

Yes, I think we should do that. I wasn't fully aware that the `()` type rust uses [is in fact just an empty tuple](https://stackoverflow.com/a/31107736/288703). That seems to make things more streamlined and avoids adding a special type to the type system (assuming that we want to have tuples anyway)